src/main/java/bio/terra/service/filedata/flight/ingest/IngestFileDirectoryStep.java

```diff
@@ -64,8 +64,8 @@ public StepResult doStep(FlightContext context) throws InterruptedException {
             // Lookup the file - on a recovery, we may have already created it, but not
             // finished. Or it might already exist, created by someone else.
             FireStoreDirectoryEntry existingEntry = fileDao.lookupDirectoryEntryByPath(dataset, targetPath);
```

Comment:
So, I guess it was decided that it was better to duplicate this call in both validate and ingest rather than use the action field to store both the `createEntry` string and the file ID?

Reply:
Yes, I thought it would make things clearer (at the expense of some duplication)!